src/rebar_fetch.erl

```diff
@@ -103,8 +104,16 @@ find_resource_module(Type, Location, Resources) ->
                     {error, io_lib:format("Cannot handle dependency ~ts.~n"
                                          "     No module for resource type ~p", [Location, Type])};
                 _ ->
+                    check_type_support(Type),
                     Type
             end;
         {Type, Module} ->
+            check_type_support(Module),
             Module
     end.
+
+check_type_support(Module) ->
+    case lists:member({check_type_support, 0},  Module:module_info(exports)) of
+        false -> ok;
+        true  -> Module:check_type_support()
+    end.
```

Comment:
We should cache that result not to call for it on every single dependency.

Reply:
the result should be cached in rebar_state ?